rails_event_store_active_record/lib/rails_event_store_active_record/mappers/yaml.rb

```diff
@@ -0,0 +1,31 @@
+module RailsEventStoreActiveRecord
+  module Mappers
+    class YAML
+      def initialize(serializer: nil)
+        @serializer = serializer || ::YAML
+      end
+
+      def build_event_payload(event)
+        {
+          id:         event.event_id,
+          data:       serializer.dump(event.data),
+          metadata:   serializer.dump(event.metadata),
+          event_type: event.class,
+        }
+      end
+
+      def build_event_entity(record)
+        return nil unless record
```

Comment:
when would that happen? why would we prefer `nil` over exception?

Reply:
its extracted from repository

Reply:
I didn't change it because I also don't know the answer

Reply:
let's drop it :)

Reply:
we can't

this is the use case you asked for
```ruby
  1) RailsEventStoreActiveRecord::EventRepository behaves like event_repository knows last event in stream
     Failure/Error:
       record.event.event_type.constantize.new(
         event_id: record.event.id,
         metadata: serializer.load(record.event.metadata),
         data:     serializer.load(record.event.data)
       )
     
     NoMethodError:
       undefined method `event' for nil:NilClass
     Shared Example Group: :event_repository called from ./spec/event_repository_spec.rb:24
```

Reply:
eventually I can move this check up, back to repository

Reply:
done